scripts/build.js

```diff
@@ -1,19 +1,25 @@
 const { execSync } = require('child_process');
-const services = require(`${__dirname}/toBuild.json`);
+//FIXME have no access to integration-content-repository. Temporary exclude it from build
+const services = require(`${__dirname}/toBuild.json`).filter(({name}) => name !== 'integration-content-repository');
 if (services && services.length > 0) {
 
+    if (process.env.DOCKER_PASSWORD && process.env.DOCKER_USERNAME) {
+        execSync('echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin', { stdio: [0, 1, 2] });
+    } else {
+        console.log('WARN: skip docker login due to missing DOCKER_PASSWORD and/or DOCKER_USERNAME environment variables');
+    }
+
     services.forEach((service)=>{
 
         console.log('We going to Build Service: ',service.name);
         console.log('The Service Version: ',service.version);
 
         execSync(`cd ${__dirname}/../services/${service.name}/ && yarn build`, { stdio: [0, 1, 2] });
-        execSync('echo $DOCKER_PASSWORD | docker login -u $DOCKER_USERNAME --password-stdin', { stdio: [0, 1, 2] });
-        execSync(`cd ${__dirname}/../services/${service.name}/ && docker build -t openintegrationhub/${service.name}:${service.version} .`, { stdio: [0, 1, 2] });
+        execSync(`cd ${__dirname}/../services/${service.name}/ && yarn run build:docker`, { stdio: [0, 1, 2], env: Object.assign({VERSION: service.version}, process.env) });
         execSync(`docker tag openintegrationhub/${service.name}:${service.version} openintegrationhub/${service.name}:latest`, { stdio: [0, 1, 2] });
 
-        execSync(`docker push openintegrationhub/${service.name}:${service.version}`, { stdio: [0, 1, 2] });
-        execSync(`docker push openintegrationhub/${service.name}:latest`, { stdio: [0, 1, 2] });
+        //execSync(`docker push openintegrationhub/${service.name}:${service.version}`, { stdio: [0, 1, 2] });
```

Comment:
If you are speaking about this changes, it also should be reverted. But I'd like to discuss build process in general: I don't think that it's good idea to set `latest` tag to any successfully built PR's docker image, because it increases risks that somebody deploys unstable version somewhere.

Reply:
That latest is only on master branch. 
The deployment will not use latest. The latest tag is just a marker for the latest image not of stable or unstable.
But as i‘m not using it feel free to remove the latest tag ;)

And i tested the Pipeline on iam without Problems.

As i said in the other post i need a big screen to Review that changes. Could be i missed something.

Reply:
Hi,
so i had a look on it. I did not see the hole change you did yesterday. 
Thats fine ! But you disable the Push to DockerHub completely. So we don't get any image to deploy to Cluster.

That Latest tag i could understand we could move that tagging to the sage comming up after Deploy in the future (integration Testing not yet implemented).

But: 
1. Why are you disabling my gcloud installation?
2. Could you please write the K8S stuff in yaml? that is more common in the Comunity.
3. you added src/ as workplace in yarn could you please name it so every partner could use /src as a lib dir ? (src/backendCommonLib or something)

Reply:
Hi, @falkvoigt.
Let me explain the reason why i did all the changes related to build process: The reason is one and simple: build process starts on every branch (including my pull request, not only master). But build process has next problems, that makes impossible for my PR to pass tests on travis
1. #93 private base image for  integration-content-repository
2. result of every build is pushed into dockerhub with latest tag (even in case of pull request). It's a) dangerous as it increases changes to deploy some incorrect b) there is no access for me to publish 
3. There is deploy phase in all travis scenario. It runs script, that just install gcloud and has commented kubectl commands. But it does not work for me, as it requires some gcloud keys in environment.
So at the moment, there is a problem, that build works only for limited set of users and this actually contradicts open-source work strategy.
Due to reasons written above I had to disable some functionality inside build scripts.

Generally I see two ways to solve all this:
1. I wait until all this issues will be fixed and then merge my PR. Then all this changes won't be required and i return back
2. Merge this PR with all this changes and then fix this issues. 
Your opinion?  

About yaml/json: I prefer json's for next reasons: we are working with javascript, so it's much easier to work with json then with yaml. If you insist on using yaml -- I can transform json descriptors into yaml, but I prefer to do this as separate task.

About src folder: Ok, I'll rename  src/lib to src/backendCommonLib

Reply:
Hi , @anton-kotenko 
the problem is relate to you fork the hole mono repo and also duplicate the travis pipeline. thats why you miss some informations. 

did you read my comment completely? 

1. yes Build breaks related to the Dockerfile fix is done in a different branch needs to be merged

2. i am fine with removing the latest tag for now until we have integration test an Production deployment. But push the build tag is needed for OIH Project.

3. That is related to your fork but OIH relies on this for deployment in test Cluster you can't get rid of it. what you can do , is make a condition relating github Projekt or simple do branching in OIH github instead of forking the hole Project.

We should go with the Community related to Kubernetes in our Github repo. So yaml is prefered Format here. Also a part of the DevOps Guideline which waiting in PR at the moment. 

Over all, the Problem is you forked the Project. We could not merge your PR aslong that commits related to google Cloud and docker push are in there. Which will disable 50% of the Build/deployment Pipeline.

Reply:
Sorry, but at this moment:
1) I can not  push new branch into original repo because of lack of privileges, so i have to fork repo, with all consequences (like broken travis pipeline)
2) My build fails in travis without this changes.
What should i do? I suppose that any other developer who is not member of this repo will face the  same problems. (is this open-source?) 

About docker push of latest/current version: I have no privileges to push that images at all in openintegrationhub repo. So If i return any push, build will just fail.

About gcloud  installation and k8s. At the moment (in master branch) script just installs gcloud cli and does nothing. So what i've commented does not changes anything in build pipeline.  

So there is one more option: I revert back all changes in build pipeline, but we ignore broken build when merging PR (and repo settings should be adjusted for this)   

> We should go with the Community related to Kubernetes in our Github repo. So yaml is prefered Format here. Also a part of the DevOps Guideline which waiting in PR at the moment.

Ok. I understand this argument.

Reply:
+1 for the open source argument @anton-kotenko. ATM it looks like @falkvoigt is the only one enjoying the privilege to build the project successfully. This sucks big time and not how an open source project must work.

Reply:
We need to give permission for you or we make a condition related on Project name or id.

Reply:
@falkvoigt this is an open source project. You would need to give the permissions to everybody on the globe.

Reply:
I can understand the fact with lag on permissions. But is it Open Source to disable Project related stuff locally and try to push it? And destroy the Pipeline which is only for OIH and not someone who wants to fork it?

it crashes on you side because you try to use Project internals. Normaly that is not the case when useing the OIH environment and not the forked one.

Reply:
not for every thing but nearly yes. I would restrict direct commiting to master. the other stuff i'm with you.

Reply:
@drobiazko i vote for open branch for every one restrict access for master. I'm not the github admin to do so. I just comment that the PR will brake pipeline on Project side.

Reply:
i already answered that.
2 different approches:
1. git branch create permission to every one and don't try to PR from a fork
2. make a condition based on Projectname/id to the push and google stuff